src/Zicht/Bundle/SolrBundle/Manager/Doctrine/Subscriber.php

```diff
@@ -34,7 +40,13 @@ public function __construct(ContainerInterface $container)
      */
     public function postPersist(LifecycleEventArgs $event)
     {
-        $this->container->get('zicht_solr.manager')->update($event->getEntity());
+        $entity = $event->getEntity();
+        if ($entity instanceof Extractable && $entity->getFile() !== null) {
+            $this->container->get('zicht_solr.manager')->extract($entity);
+            return;
+        }
+
+        $this->container->get('zicht_solr.manager')->update($entity);
     }
```

Comment:
The code is exactly the same as preUpdate, wouldn`t it make sense to wrap that in function?